src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -394,6 +400,20 @@ static PrivateKey privateKey(Session session, long keyID, String algorithm,
         boolean keySensitive = (attrs[0].getBoolean() ||
                 attrs[1].getBoolean() || !attrs[2].getBoolean());
 
+        if ((SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm) && keySensitive) {
```

Comment:
These tests should be reordered least-expensive first:
```
    keySensitive && (SunPKCS11.mysunpkcs11 != null) && "RSA".equals(algorithm)
```

Reply:
Code updated according to code review.